itest/lnd_multi-hop_test.go

```diff
@@ -297,6 +371,7 @@ func runMultiHopReceiverChainClaim(ht *lntest.HarnessTest,
 		Value:      invoiceAmt,
 		CltvExpiry: finalCltvDelta,
 		Hash:       payHash[:],
+		RouteHints: routeHints,
```

Comment:
Initially I thought we could pass `Private: true` instead since `AddHoldInvoice` will construct the route hints for us. Turns out Bob has no public channels, so he is not considered as a public node, hence Carol won't add route hints because of this check,
https://github.com/lightningnetwork/lnd/blob/c6a68d193c56297b776634a28751aee7b485602b/lnrpc/invoicesrpc/addinvoice.go#L515-L526